primitives/src/merkle_tree/examples.rs

```diff
@@ -16,35 +19,37 @@ use sha3::{Digest, Sha3_256};
 use typenum::U3;
 
 /// Wrapper for rescue hash function
-pub struct RescueHash<F: RescueParameter> {
+pub struct RescueHash<I: Index, F: RescueParameter> {
+    phantom_i: PhantomData<I>,
     phantom_f: PhantomData<F>,
 }
 
-impl<F: RescueParameter> DigestAlgorithm<F, u64, F> for RescueHash<F> {
+impl<I: Index, F: RescueParameter + From<I>> DigestAlgorithm<F, I, F> for RescueHash<I, F> {
     fn digest(data: &[F]) -> F {
         let perm = Permutation::default();
         perm.sponge_no_padding(data, 1).unwrap()[0]
     }
 
-    fn digest_leaf(pos: &u64, elem: &F) -> F {
-        let data = [F::from(*pos), *elem, F::zero()];
+    fn digest_leaf(pos: &I, elem: &F) -> F {
+        let data = [F::from(pos.clone()), *elem, F::zero()];
         let perm = Permutation::default();
         perm.sponge_no_padding(&data, 1).unwrap()[0]
     }
 }
 
 /// A standard merkle tree using RATE-3 rescue hash function
-pub type RescueMerkleTree<F> = MerkleTree<F, RescueHash<F>, u64, U3, F>;
+pub type RescueMerkleTree<F> = MerkleTree<F, RescueHash<u64, F>, u64, U3, F>;
 
+impl Index for BigUint {}
 /// Example instantiation of a SparseMerkleTree indexed by BigUInt
-pub type SparseMerkleTree<E, F> = MerkleTree<E, RescueHash<F>, BigUint, U3, F>;
+pub type SparseMerkleTree<E, F> = UniversalMerkleTree<E, RescueHash<BigUint, F>, BigUint, U3, F>;
```

Comment:
Why the `Index` type is `BigUint` instead of `F`?

(plus `Field` impl `Copy`)

Reply:
Good point. I don't know before that `Field` already bound for various Ops. Let me work on it.